hibernate-core/src/main/java/org/hibernate/dialect/DB2400V7R1Dialect.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.dialect;
+
+import org.hibernate.dialect.unique.DefaultUniqueDelegate;
+import org.hibernate.dialect.unique.UniqueDelegate;
+
+/**
+ * An SQL dialect for i. This class provides support for DB2 Universal Database for i V7R1 and
+ * later, also known as DB2/400.
+ *
+ * @author Pierrick Rouxel (pierrickrouxel)
+ */
+public class DB2400V7R1Dialect extends DB2Dialect {
```

Comment:
I'm surprised it does not extend `DB2400Dialect`?

Reply:
It's voluntary because the real function of DB2400Dialect is emulate the sequences. But maybe I should have extend DB2400Dialect and revert the feature for a better understanding ?

Reply:
Agreed there is the sequence support and a lot of noise about limits that should be removed considering we have a specific limit handler but there is at least these methods which are different from the `DB2Dialect`:
```
	@Override
	public String getForUpdateString() {
		return " for update with rs";
	}

	@Override
	public IdentityColumnSupport getIdentityColumnSupport() {
		return new DB2390IdentityColumnSupport();
	}
```

Reply:
For the DB2390IdentityColumnSupport V7R1 supports VALUES clause. There is no reason to continue to use sysibm.sysdummy1.

I copied for update with rs.

Reply:
I also replaced the unique delegate because the V7R1 supports unique constraint on column which has a different behaviour of the unique index with null. It makes unique compatible with Postgres, MySQL...